Mathlib/NumberTheory/DirichletCharacter/Basic.lean

```diff
@@ -0,0 +1,69 @@
+/-
+Copyright (c) 2023 Ashvni Narayanan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ashvni Narayanan, Moritz Firsching
+-/
+import Mathlib.NumberTheory.LegendreSymbol.MulCharacter
+import Mathlib.Data.ZMod.Algebra
+
+/-!
+# Dirichlet Characters
+
+Let `R` be a monoid. A Dirichlet character `χ` of level `n` over `R` is a homomorphism from the unit
+group `(ZMod n)ˣ` to `Rˣ`. We then obtain some properties of `ofUnitHom χ`.
+
+Main definitions:
+
+- `DirichletCharacter`: The type representing a Dirichlet character.
+
+## TODO
+
+- `change_level`: Extend the Dirichlet character χ of level `n` to level `m`, where `n` divides `m`.
+- definition of conductor
+
+## Tags
+
+dirichlet character, multiplicative character
+-/
+-- TODO: move to Data.ZMod.Basic?!
+@[simp]
+lemma cast_hom_self {n : ℕ} : ZMod.castHom dvd_rfl (ZMod n) = RingHom.id (ZMod n) :=
+  RingHom.ext_zmod (ZMod.castHom dvd_rfl (ZMod n)) (RingHom.id (ZMod n))
+
+/-- The Dirichlet character of level `n`. -/
```

Comment:
Better "The type of Dirichlet characters of level `n`".
(The current version sounds like there is exactly one Dirichlet character of level `n`, which is `DirichletCharacter R n`.)

Reply:
Yes, much better, done